From 3ebcfb242e562d09689f821fd7a780baa65e3899 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Fri, 7 Aug 2015 19:07:59 +0800
Subject: [PATCH] MLK-11333 ARM: imx: enable busfreq daemon by default

By default, busfreq mode is set to high bus mode,
if there is no high speed device enabled, system
should be able to enter low bus mode, that is why
we have a daemon thread created to handle such
scenario, this daemon should be enabled after busfreq
driver probed.

However, we missed to enable it before, this patch
enables it.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/busfreq-imx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-imx/busfreq-imx.c b/arch/arm/mach-imx/busfreq-imx.c
index 2408e4d..2cd1e59 100644
--- a/arch/arm/mach-imx/busfreq-imx.c
+++ b/arch/arm/mach-imx/busfreq-imx.c
@@ -1261,6 +1261,10 @@ static int busfreq_probe(struct platform_device *pdev)
 	register_pm_notifier(&imx_bus_freq_pm_notifier);
 	register_reboot_notifier(&imx_busfreq_reboot_notifier);
 
+	/* enter low bus mode if no high speed device enabled */
+	schedule_delayed_work(&bus_freq_daemon,
+		msecs_to_jiffies(10000));
+
 	/*
 	 * Need to make sure to an entry for the ddr freq change code address in the IRAM page table.
 	 * This is only required if the DDR freq code and suspend/idle code are in different OCRAM spaces.
-- 
1.8.0

